File: Dot examples/ASP Book Examples/Book Examples/Chapter25/VB/App_Code/NewsRotator.vb
Imports System
Imports System.Data
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Collections.Generic

Namespace myControls
    Public Class NewsRotator
        Inherits WebControl

        Public Shared Function GetNews(ByVal context As HttpContext) As String
            Dim News As New List(Of String)()
            News.Add("Martians attack!")
            News.Add("Moon collides with earth!")
            News.Add("Life on Jupiter!")

            Dim rnd As Random = New Random()
            Return News(rnd.Next(News.Count))
        End Function

        Protected Overrides Sub RenderContents(ByVal writer As HtmlTextWriter)
            Context.Response.WriteSubstitution(AddressOf GetNews)
        End Sub

    End Class
End Namespace

File: Dot examples/ASP Book Examples/Book Examples/Chapter25/CS/App_Code/NewsRotator.cs
using System;
using System.Data;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Collections.Generic;

namespace myControls
{
    public class NewsRotator : WebControl
    {

        public static string GetNews(HttpContext context)
        {
            List<String> news = new List<string>();
            news.Add("Martians attack!");
            news.Add("Moon collides with earth!");
            news.Add("Life on Jupiter!");

            Random rnd = new Random();
            return news[rnd.Next(news.Count)];
        }

        protected override void RenderContents(HtmlTextWriter writer)
        {
            Context.Response.WriteSubstitution(GetNews);
        }

    }
}